### PR TITLE
Two files added of linked list

### PR DESCRIPTION
**Pull request by rishabhgarg25699**
Implementation of linked list
and second is implementation of linked list using header node

### CODE REVIEW

**Comment by sahilbansal17** [comment]
@rishabhgarg25699 You have added files in .txt format. You need to change it to .cpp!
Also, name the files properly. 
Follow the [Coding Standards](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards) as well.

File: Data Structures/Linked-Lists/All implementations of linked list.txt [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+{
+	int info;
+	struct node *link;
+};
+
+struct node *create_list(struct node *start);
+void display(struct node *start);
+void count(struct node *start);
+void search(struct node *start,int data);
+struct node *addatbeg(struct node *start,int data);
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/All implementations of linked list.txt]: Follow [Coding Standards](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards). Use loweCamelCase for function name. Fix this in all files

File: Data Structures/Linked-Lists/All implementations of linked list.txt [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+		 case 12:
+			 exit(1);
+		 default:
+			 printf("Wrong choice\n");
+		}/*End of switch */
+	}/*End of while */
+}/*End of main()*/
+
+
+struct node *create_list(struct node *start)
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/All implementations of linked list.txt]: Give a small explanation of your function. Fix this in complete code

**Comment by rishabhgarg25699** [comment]
Now see, I changed my file according to rules and regulations.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+#include<stdio.h>
+#include<stdlib.h>
+
+struct node
+{
+	int info;
+	struct node *link;
+};
+
+struct node *create_List(struct node *start);
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: This should be `createList(arguments)`.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+	}/*End of while */
+
+return 0;
+
+}
+
+/*End of main()*/
+
+
+struct node *create_List(struct node *start)
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: Add a brief explanation of what this function does. Fix it in the complete code.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+		ptr->link=prev;
+		prev=ptr;
+		ptr=next;
+	}
+	start=prev;
+	return start;
+}
+
+/*End of reverse()*/
+
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: Remove unnecessary blank lines.

Comment (rishabhgarg25699) [reply Data Structures/Linked-Lists/Implementations of single linked list.cpp]: No,By blanks it is clear to understand the program easily.
Blank make the program easily understandable to other .

**Comment by aditya81070** [comment]
@rishabhgarg25699 you deleted the file. Now, what should we review?

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+		printf("Enter the element to be inserted : ");
+		scanf("%d",&data);
+		start=addAtEnd(start,data);	
+	}
+	return start;
+}
+
+/*End of create_list()*/
+
+
```
Comment (ramitsawhney27) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: Keep a uniform number of blank lines. 1 after a function ends, 1 for the comment and 1 following it. Remove the extra from everywhere.

**Comment by rishabhgarg25699** [comment]
Now please review my code one more time. Please pass this time.
I will implement all your rules in next file in proper manner.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp (hunk 6 above)
Comment (rishabhgarg25699) [reply Data Structures/Linked-Lists/Implementations of single linked list.cpp]: I remove all blanks.No blanks for function end and no space for comment.

**Comment by rishabhgarg25699** [comment]
What happen
why merging is not haooening???

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+		ptr->link=prev;
+		prev=ptr;
+		ptr=next;
+	}
+	start=prev;
+	return start;
+}
+/*End of reverse()*/
+
+/*  In whole program there is only one loop many times in different block os time complexity will be k*O(n)=O(n)  */
```
Comment (sahilbansal17) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: Please explain the **time complexity for each of the functions** separately as comments above the function. The time complexity of the entire program is irrelevant.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+struct node *addBefore(struct node *start,int data,int item)
+{
+	struct node *tmp,*p;
+	if(start==NULL )
+	{
+		printf("List is empty\n");
+		return start;
+	}	
+	/*If data to be inserted before first node*/
+	if(item==start->info)
```
Comment (sahilbansal17) [inline Data Structures/Linked-Lists/Implementations of single linked list.cpp]: You can directly call the **addAtBeg** function here for reusability of the code.

**Comment by sahilbansal17** [comment]
@rishabhgarg25699 We are really sorry for late review. There were a lot of PRs, we will try to be quicker next time.

**Comment by ramitsawhney27** [comment]
You have NOT made the change I asked you to do. Add a pointer maintaining the tail of the LL to reduce all end insertion operations to O(1). This PR won't pass otherwise.

**Comment by sahilbansal17** [comment]
@ramitsawhney27 There are two PRs going on for the same issue, somehow, both of them started working.
#51 is where you requested this change. So, @rishabhgarg25699 is not aware of this.

**Comment by ramitsawhney27** [comment]
@sahilbansal17 who began it first then? As the other PR is complete and correct now, I believe the points should be given for it and @rishabhgarg25699 can move onto a different issue.

**Comment by sahilbansal17** [comment]
@ramitsawhney27 The other PR is also incomplete.

File: Data Structures/Linked-Lists/Implementations of single linked list.cpp (hunk 7 above)
Comment (sahilbansal17) [reply Data Structures/Linked-Lists/Implementations of single linked list.cpp]: @rishabhgarg25699 Please take this also into consideration.

**Comment by sahilbansal17** [comment]
@rishabhgarg25699 Still there are unresolved changes. See above.

**Comment by ramitsawhney27** [comment]
@sahilbansal17 we'll accept the one that gets done first then.

**Comment by rishabhgarg25699** [comment]
@sahilbansal17 Please tell me fast what are the unresolved issues.
I have done all issues clear.

Add a pointer maintaining the tail of the LL to reduce all end insertion operations to O(1).
THIS ISSSUES IN NOT FOR ME 
I THINK SO

**Comment by sahilbansal17** [comment]
@rishabhgarg25699 You have to do it. Maintain a pointer for tail of the linked list and reduce the end operations to `O(1)`.

**Comment by sahilbansal17** [comment]
@rishabhgarg25699 Please let us know if you are still working, otherwise the PR must be closed.

**Comment by rishabhgarg25699** [comment]
I an unable to do the problem i.e Add a pointer maintaining the tail of the LL to reduce all end insertion operations to O(1).
If you help me to solve this then i can do it otherwise please close it.

**Pull request closed without merging**